usaspending_api/common/helpers/s3_helpers.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+import io
+import logging
+import re
+
+from typing import Optional, List
+
+from usaspending_api import settings
+
+logger = logging.getLogger("script")
+
+
+def access_s3_object_list(
+    bucket_name: str, regex_pattern: Optional[str] = None
+) -> List["boto3.resources.factory.s3.ObjectSummary"]:
+    """Find all S3 objects in provided bucket.
+
+    If regex is passed, only keys which match the regex are returned
+    """
+
+    logger.info("Gathering all deleted transactions from S3")
+    try:
+        bucket = get_s3_bucket(bucket_name=bucket_name)
+        bucket_objects = list(bucket.objects.all())
+    except Exception:
+        msg = (
+            "Verify settings.USASPENDING_AWS_REGION and settings.DELETED_TRANSACTION_JOURNAL_FILES are correct "
+            "or env variables: USASPENDING_AWS_REGION and DELETED_TRANSACTION_JOURNAL_FILES are set"
+        )
+        logger.exception(msg)
+        bucket_objects = None
```

Comment:
Should this be re-raising the exception?  Currently it is swallowing it.  When I run locally it throws up all over my screen but keeps running as if nothing happened.

It sort of looks like it's intentional but I'm not sure I understand why.  If it can't access the S3 isn't that a problem?

Reply:
It is intentional. I'm open to a discussion about it

Reply:
"...throws up all over my screen" == stack trace in log? If the stack trace is unwelcome it can easily be removed :-)

Reply:
haha... that wasn't the point.  I was just concerned about continuing after not being able to access the S3.  If we can't access the S3 we can't pick up delete files.  If we can't pick up delete files then there's a pretty good chance some will be skipped on the next run since it's driven by date.

Reply:
I understood your concerns. So either:
A: Cannot run the loader without a working connection to an S3 bucket
B: Re-implement the local mode

Reply:
Added the snippet below to `usaspending_api/transactions/transaction_delete_journal_helpers.py` so the inability to connect to S3 stops the script
```python
    ...
        if objects is None:
            raise Exception("Problem accessing S3 for deleted records")
    ...
```

Reply:
This will fail if a delete file is not created that night for whatever reason.  Looking through the S3 bucket, I see a couple gaps in the delete file dates, specifically 1/18/20 and 2/22/20.

Reply:
1. The behavior should be different between an exception and no files returned (`None` vs empty list `[]`)
2. The current FABS only uploads a file when there are delete records while the new scripts will always create a file. It will be empty in the event no records were deleted in the timeframe which tracks with FPDS delete files

Reply:
> The behavior should be different between an exception and no files returned

Yup.  You're right.  My bad.